tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -164,8 +164,27 @@ namespace tcf {
             "Unsupported payload format found in the input");
 
         // Decrypt Encryption key
-        ByteArray encrypted_session_key_bytes = HexStringToBinary(encrypted_session_key);
-        session_key = enclaveData->decrypt_message(encrypted_session_key_bytes);
+        ByteArray encrypted_session_key_bytes = \
+            HexStringToBinary(encrypted_session_key);
+
+        WorkOrderPreProcessedKeys wo_pre_proc_keys;
+        if (this->enclave_type == WPE_ENCLAVE) {
```

Comment:
Please subclass `WorkOrderProcessor` for WPE specific functionalities.

Reply:
Done